src/Npgsql/TypeHandlerRegistry.cs

```diff
@@ -595,18 +595,14 @@ internal bool TryGetByOID(uint oid, out TypeHandler handler)
                 if (ByType.TryGetValue(type, out handler))
                     return handler;
 
-                Type arrayElementType = null;
+                PostgresType postgresType;
```

Comment:
With how we now identify `IList<>`, we need to handle CLR types first. Types like `byte[]` and `ArraySegment` implement `IList<T>` and would thus be handled as arrays rather than by their "native" handlers.